File: src/main/java/me/arasple/mc/trmenu/action/acts/ActionOpen.java
package me.arasple.mc.trmenu.action.acts;

import io.izzel.taboolib.util.ArrayUtil;
import me.arasple.mc.trmenu.TrMenu;
import me.arasple.mc.trmenu.action.base.AbstractAction;
import me.arasple.mc.trmenu.api.TrMenuAPI;
import me.arasple.mc.trmenu.menu.Menu;
import me.arasple.mc.trmenu.utils.Vars;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;

/**
 * @author Arasple
 * @date 2019/12/28 19:07
 */
public class ActionOpen extends AbstractAction {

    @Override
    public String getName() {
        return "gui|trmenu|menu|open";
    }

    @Override
    public void onExecute(Player player) {
        String content = getContent(player);
        String menuName = content.split(" ")[0];
        Menu menu = TrMenuAPI.getMenu(menuName);
        if (menu != null) {
            Bukkit.getScheduler().runTask(TrMenu.getPlugin(), () -> menu.open(player, true, ArrayUtil.arrayJoin(content.split(" "), 1).split(" ")));
        } else {
            player.sendMessage(ChatColor.RED + "Menu " + ChatColor.YELLOW + menuName + ChatColor.RED + " does not exist");
        }
    }

}

File: src/main/kotlin/me/arasple/mc/trmenu/modules/action/impl/menu/ActionOpen.kt
package me.arasple.mc.trmenu.modules.action.impl.menu

import io.izzel.taboolib.internal.apache.lang3.math.NumberUtils
import me.arasple.mc.trmenu.api.TrMenuAPI
import me.arasple.mc.trmenu.api.events.MenuOpenEvent
import me.arasple.mc.trmenu.data.MetaPlayer.setArguments
import me.arasple.mc.trmenu.modules.action.base.Action
import me.arasple.mc.trmenu.utils.Tasks
import org.bukkit.entity.Player

/**
 * @author Arasple
 * @date 2020/3/29 21:36
 */
class ActionOpen : Action("open(s)?|(open)?(-)?gui|(tr)?menu") {

    override fun onExecute(player: Player) {
        Tasks.delay(1) {
            val args = getContentSplited(player, " ")
            val split = args[0].split(":")
            val menu = TrMenuAPI.getMenuById(split[0]) ?: return@delay
            val page = if (split.size > 1) NumberUtils.toInt(split[1], -1) else -1

            player.setArguments(if (args.size == 1) arrayOf() else args.toTypedArray().copyOfRange(1, args.size - 1))
            menu.open(player, page, MenuOpenEvent.Reason.PLAYER_COMMAND)
        }
    }

}